src/utils/codemods/findAvailableCodemodsInNodeModules.js

```diff
@@ -2,32 +2,115 @@ const log = require('@dhis2/cli-helpers-engine').reporter
 const fs = require('fs-extra')
 const path = require('path')
 
-const isNotDotFolder = value => value !== '.' && value !== '..'
+const isDotFolder = name => name[0] === '.'
+
+const isNotDotFolder = name => !isDotFolder(name)
 
 const isDotJSFile = name => new RegExp('[.]js$').test(name)
 
-const extractCodemods = (paths, moduleName) => {
-    const { DHIS2_NODE_MODULES } = paths
-    const codemodsPath = path.join(DHIS2_NODE_MODULES, moduleName, 'codemods')
+const isFolder = path => {
+    if (!fs.existsSync(path)) return false
+    return fs.lstatSync(path).isDirectory()
+}
+
+const hasCodemods = codemodsPath =>
+    !!fs.readdirSync(codemodsPath).filter(curFile => {
+        const curPath = path.join(codemodsPath, curFile)
+
+        if (isFolder(curPath)) return false
+        if (!curFile.match(/\.(j|t)sx?$/)) return false
+
+        return true
+    }).length
 
-    if (!fs.pathExistsSync(codemodsPath)) {
-        log.debug(
-            `Cound not find a codemods folder in node module '${moduleName}' directory`
+const getPathOfDependency = dependency => {
+    const packageJsonResolvePath = `${dependency}/package.json`
+
+    try {
+        const resolvedPath = require.resolve(
+            packageJsonResolvePath,
+            process.env.NODE_ENV === 'test'
+                ? { paths: [process.env.RESOLVE_PATH] }
+                : undefined
```

Comment:
Should this always be `paths.cwd`?  That way it resolves relative to wherever the command is run (even if cwd is passed on the command-line) and we don't need a special case for tests

Reply:
In normal use cases it needs to be `${paths.cwd}/node_modules`, when testing, it needs to be `${paths.cwd}/test/node_modules` for the integration tests and  `src/utils/codemods/__test__/findAvailableCodemodsInNodeModules/` for a unit test

Reply:
I did it this way because you can't mock `require.resolve` with jest

Reply:
ping @amcgee

Reply:
@Mohammer5 - it's common practice in our cli modules to support a `--cwd path/to/cwd` argument - would be nice to support this (falling back to `process.cwd()`) and pass that to this function, then you can just pass a `cwd` argument to this function in jest and don't need any test-specific code.

Reply:
In this case this has nothing to do with the "normal" use case, it's for testing only.

The normal use-case is: The `findAvailableCodemodsInNodeModules` function uses the `--cwd` option (defaults to `process.cwd`) (see [here](https://github.com/dhis2/cli-utils-codemods/blob/6374f2d53903188323aef4113121b93cb3b36e0b/src/commands/list.js#L23)) to locate the `package.json`.
The function then extracts the dependencies from the package.json so it can iterate through the found packages to find only those that contain codemods. In order to find the correct location, it uses `require.resolve`.

When testing this function with jest, I need to have some modules with codemods in the `node_modules` folder, which is not guaranteed unless when providing it in a mocked node_modules folder, hence the reason to mock `require.resolve`.

You can't mock `require.resolve`, which is the reason for the check

Reply:
@amcgee and I had a quick chat on slack and the issue was that `require.resolve` won't respect the `--cwd` option when passed in.

This has been addressed and there's no different value for test environments anymore. Instad the cwd is passed as a path to the `resolve` method.